### PR TITLE
nixosTests.containers*: port to python

### PR DESCRIPTION
**Pull request by tfc**
###### Motivation for this change

#72828

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @worldofpeace @flokli @Lassulus

### CODE REVIEW

**Comment by tfc** [comment]
@GrahamcOfBorg test containers-ephemeral containers-hosts containers-reloadable containers-tmpfs

File: nixos/tests/containers-reloadable.nix [hunk 1]
```diff
@@ -48,18 +48,25 @@ in {
     c1System = nodes.client_c1.config.system.build.toplevel;
     c2System = nodes.client_c2.config.system.build.toplevel;
   in ''
-    $client->start();
-    $client->waitForUnit("default.target");
-    $client->succeed("[[ \$(nixos-container run test1 cat /etc/check) == client_base ]] >&2");
+    client.start()
+    client.wait_for_unit("default.target")
+
+    client.succeed("[[ $(nixos-container run test1 cat /etc/check) == client_base ]] >&2")
```
Comment (flokli) [inline nixos/tests/containers-reloadable.nix]:
```suggestion
    assert "client_base" in client.succeed("nixos-container run test1 cat /etc/check")
```

**Review by flokli — COMMENTED** [review]
one minor nitpick.

**Comment by flokli** [comment]
@GrahamcOfBorg test containers-ephemeral containers-hosts containers-reloadable containers-tmpfs

**Pull request merged**